docs/platform-integration/invoke-platform-code.md

```diff
@@ -0,0 +1,144 @@
+
+The project for a .NET MAUI app contains a _Platforms_ folder, with each child folder representing a platform that .NET MAUI can target:
+
+:::image type="content" source="media/invoke-platform-code/platform-folders.png" alt-text="Platform folders screenshot.":::
+
+The folders for each target platform contain platform-specific code that starts the app on each platform, plus any additional platform code you add. At build time, the build system only includes the code from each folder when building for that specific platform. For example, when you build for Android the files in the _Platforms_ > _Android_ folder will be built into the app package, but the files in the other _Platforms_ folders won't be. This approach uses a feature called multi-targeting to target multiple platforms from a single project.
+
+Multi-targeting can be combined with partial classes and partial methods to invoke native platform functionality from cross-platform code. The process for invoking platform code from cross-platform code is to:
+
+1. Define the cross-platform API as a partial class that defines partial method signatures for any operations you want to invoke on each platform. For more information, see [Define the cross-platform API](#define-the-cross-platform-api).
+1. Implement the cross-platform API per platform, by defining the same partial class and the same partial method signatures, while also providing the method implementations. For more information, see [Implement the API per platform](#implement-the-api-per-platform).
+1. Invoke the cross-platform API by creating an instance of the partial class and invoking its methods as required. For more information, see [Invoke the cross-platform API](#invoke-the-cross-platform-api).
+
+## Define the cross-platform API
+
+To invoke platform code from cross-platform code, the first step is to define the cross-platform API as a [partial class](/dotnet/csharp/programming-guide/classes-and-structs/partial-classes-and-methods#partial-classes) that defines [partial method](/dotnet/csharp/programming-guide/classes-and-structs/partial-classes-and-methods#partial-classes) signatures for any operations you want to invoke on each platform.
+
+The following example shows a cross-platform API that can be used to retrieve the orientation of a device:
+
+```csharp
+namespace CallPlatformCodeDemos.Services
+{
+    public enum DeviceOrientation
+    {
+        Undefined,
+        Landscape,
+        Portrait
+    }
+
+    public partial class DeviceOrientationService
+    {
+        public partial DeviceOrientation GetOrientation();
+    }
+}
+```
+
+The partial class is named `DeviceOrientationService`, which includes a partial method named `GetOrientation`.
+
+## Implement the API per platform
+
+After defining the cross-platform API, it must be implemented on your required platforms by defining the same partial class and the same partial method signatures, while also providing the method implementations.
```

Comment:
Maybe emphesize that it always needs a (dummy) declaration. Even if only 1 platform, for some reason, needs an actual implementation?

Reply:
Does it? I *think* this was fine when I wrote it - that MAUI just builds for the target platform, rather than all four platforms at once (which was the behaviour until recently). I'll double check.

Reply:
I just double checked. You only need to provide a dummy/implementation on the platform you're building for. So if you only ever build for iOS/Android, no need to provide Catalyst/Windows dummies.

This was in the current public preview (with no implicit usings).